homeassistant/components/wevolor/cover.py

```diff
@@ -0,0 +1,80 @@
+"""Cover component for shades controlled by the Wevolor controller."""
+
+from __future__ import annotations
+
+from pywevolor import Wevolor
+
+from homeassistant.components.cover import (
+    SUPPORT_CLOSE,
+    SUPPORT_CLOSE_TILT,
+    SUPPORT_OPEN,
+    SUPPORT_OPEN_TILT,
+    SUPPORT_STOP,
+    SUPPORT_STOP_TILT,
+    CoverDeviceClass,
+    CoverEntity,
+)
+
+from .const import CONFIG_CHANNELS, CONFIG_TILT, DOMAIN
+
+
+async def async_setup_entry(hass, config_entry, async_add_entities):
+    """Set up the Wevolor shades."""
+
+    wevolor = hass.data[DOMAIN][config_entry.entry_id]
+
+    entities = [
+        WevolorShade(wevolor, i, config_entry.data[CONFIG_TILT])
+        for i in range(1, config_entry.data[CONFIG_CHANNELS] + 1)
+    ]
+    async_add_entities(entities, True)
+
+
+class WevolorShade(CoverEntity):
+    """Cover entity for control of Wevolor remote channel."""
+
+    _attr_assumed_state = True
+    _wevolor: Wevolor
+    _channel: int
```

Comment:
This is already implied by https://github.com/home-assistant/core/pull/63087/files#diff-817a8b1e483d7cf98d65c509fbebffb947a05f000f9f5c7a543ee634652d463cR40-R43
```suggestion
```

Reply:
Question about this, as using type hints in python is new to me -- I had assumed it'd be helpful to include type hints explicitly for these class attributes, since otherwise they're not strictly typed. Does that just not matter here because they're private attributes? Or is the interpreter able imply the type from the place they're first initialized?